TaskList/TaskList/TaskListHelper.cs

```diff
@@ -0,0 +1,231 @@
+                {
+                    Console.WriteLine(read.ReadToEnd());
+                }
+            }
+
+            catch
+            {
+                Console.WriteLine("Error");
+            }
+        }
+
+        public void ViewPlans()
+        {
+            for (int i = 0; i < count; i++)
+            {
+                ColorDisplay(6);
+                TaskList task = taskList[i];
+                RegularTask taskDate = new RegularTask();
+                task?.GetAlarm(taskDate.Date);
+                Console.WriteLine(task?.TaskConvert(i));
+                ColorDisplay(6);
+            }
+        }
+
+        public void AddNewTask()
+        {
+            Console.WriteLine("Input you task name; task date; task time; task priority;");
+
+            if (count > 19)
+            {
+                Console.WriteLine("Error. Task limit is over");
+                return;
+            }
+
+            var inputDate = Console.ReadLine();
+            var task = ParseNewTask(inputDate);
+            WriteTask(task);
+        }
+
+        public TaskList ParseNewTask(string input)
+        {
+            string[] parts = input?.Split(';');
+
+            if (parts.Length < 1 || parts.Length > 4)
+            {
+                Console.WriteLine("input error. Try input correctly or press alt f4 ");
+                return null;
+            }
+
+            return MakeNewTask(parts);
+        }
+
+        private DateTime ParseDate(string inputString, string dateFormat)
+        {
+            DateTime parsedInput;
+            while (!DateTime.TryParseExact(inputString, dateFormat, CultureInfo.InvariantCulture, DateTimeStyles.None, out parsedInput))
+            {
+                dateFormat = default;
+                break;
+            }
+            return parsedInput;
+        }
+
+        private TaskList MakeNewTask(string[] parts)
+        {
+            switch (parts.Length)
+            {
+                case 1:
+                    return AddTaskOnlyWithName(parts);
+                case 2:
+                    return AddTasksWithNameAndDate(parts);
+                case 3:
+                    return AddTasksWithNameDateTime(parts);
+                case 4:
+                    return AddAllTasksParametrs(parts);
+                default:
+                    return default;
+            }
+        }
+
+        private TaskList AddTaskOnlyWithName(string[] parts)
+        {
+            TaskList task = new PriorityTask(parts[0]);
+            return task;
+        }
+
+        private TaskList AddTasksWithNameAndDate(string[] parts)
+        {
+            TaskList task = new RegularTask(parts[0], ParseDate(parts[1], "dd.MM.yyyy"));
+            return task;
+        }
+
+        private TaskList AddTasksWithNameDateTime(string[] parts)
+        {
+            TaskList task = new RegularTask(parts[0], ParseDate(parts[1], "dd.MM.yyyy"), ParseDate(parts[2], "HH:mm"));
```

Comment:
extract formats into 2 constants and re-use
```suggestion
            TaskList task = new RegularTask(parts[0], ParseDate(parts[1], DateFormat), ParseDate(parts[2], TimeFormat));
```